src/compiler/crystal/macros/methods.cr

```diff
@@ -158,6 +158,31 @@ module Crystal
       end
     end
 
+    def interpret_constant(arg, type, name)
+      case type = type.lookup_path self.parse_path(arg, name)
+      when Const
+        type.value
+      when Type
+        TypeNode.new(type)
+      else
+        NilLiteral.new
+      end
+    end
+
+    def interpret_has_constant?(arg, type, name)
+      BoolLiteral.new !!type.lookup_path self.parse_path(arg, name)
```

Comment:
```suggestion
      case type = type.lookup_path parse_path(arg, name)
      when Const
        type.value
      when Type
        TypeNode.new(type)
      else
        NilLiteral.new
      end
    end

    def interpret_has_constant?(arg, type, name)
      BoolLiteral.new !!type.lookup_path parse_path(arg, name)
```

Reply:
Is the suggestion here to just remove `self.` or? The diff is quite confusing.

Reply:
Yeah, that's it. Sorry, should've been more clear.